# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#    Configuration file for the HoloLens 2.
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

# Switches for whether short throw and/or long throw depth frames should be used for calculating the point cloud.
useShortThrow: false
useLongThrow: true

# Hyper parameters used for median filtering. Only affects the InitialDepthDataReceiver.
doMedianFiltering: true
medianFilterWindowSize: 3

# Hyper parameters used for downsampling.
doDownsampling: true
downsamplingLeafSize: 0.01
downsamplingMinPointsPerVoxel: 1 # Currently, this only affects the FasterDepthDataReceiver. This parameter was just meant to test something real quick, but changing it to a larger value than 1 didn't result in the desired effects.

# Hyper parameters used for outlier removal with a radius outlier removal filer. Only affects the InitialDepthDataReceiver.
doOutlierRemovalRadius: false
outlierRemovalRadiusSearch: 0.05
outlierRemovalMinNeighborsInRadius: 15

# Hyper parameters used for outlier removal with a statistical outlier removal filter. Only affects the InitialDepthDataReceiver.
doOutlierRemovalStatistical: false
outlierRemovalNeighborsToCheck: 10
outlierRemovalStdDeviationMultiplier: 0.5

# Hyper parameters uesd for outlier removal by clustering and discarding too small clusters. Only affects the InitialDepthDataReceiver.
doOutlierRemovalClustering: true
outlierRemovalClusterTolerance: 0.08
outlierRemovalMinClusterSize: 600

# Hyper parameters for clustering of similar pixels in received depth maps. Only affects the FasterDepthDataReceiver.
pixelClusteringUsePcl: true
pixelClusteringNeighboringPixelDistance: 2.5
pixelClusteringAbsoluteDepthValueSimilarity: 50 # A value of 50 indicates that neighboring pixels must have a depth value within a range of +/- 5 centimeters of the current pixel's depth value in order for them to be considered in the same cluster. 
pixelClusteringMinClusterSize: 100

# Sensor intrinsics of the short throw depth sensor.
shortThrowMinDepth: 0.1
shortThrowMinReliableDepth: 0.2
shortThrowMaxReliableDepth: 1.0
shortThrowMaxDepth: 1.0

# Sensor intrinsics of the long throw depth sensor.
longThrowMinDepth: 0.2
longThrowMinReliableDepth: 0.75 # While the sensor generally seems to provide reliable values even for values below 0.75 meters, there are sometimes a lot of ghost points in that range.
longThrowMaxReliableDepth: 8.0  # Sometimes objects are reported as being closer than they actually are for depth values above ~8.0 meters.
longThrowMaxDepth: 16.0

# Switches and sensor intrinsics related to discarding noisy pixels of the depth sensors with a rectangular region of
# interst. Pixels outside that region won't be used in the creation of a point cloud from a sensor frame. Positions and
# sizes are relative positions and sizes in the depth sensor's image space. Only affects the InitialDepthDataReceiver.
discardNoisyPixelsRect: true
noisyPixelRemovalRectCenterX: 0.5
noisyPixelRemovalRectCenterY: 0.5
noisyPixelRemovalRectWidth: 0.8
noisyPixelRemovalRectHeight: 0.8

# Switches and sensor intrinsics related to discarding noisy pixels of the depth sensors with a circular region of
# interest. Pixels outside that region won't be used in the creation of a point cloud from a sensor frame. Positions and
# sizes are relative positions and sizes in the depth sensor's image space. Only affects the InitialDepthDataReceiver.
discardNoisyPixelsCircle: true
noisyPixelRemovalCircleCenterX: 0.5
noisyPixelRemovalCircleCenterY: 0.5
noisyPixelRemovalCircleRadius: 0.45